java/memory/memory-core/src/test/java/org/apache/arrow/memory/DefaultAllocationManagerFactory.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.memory;
+
+import org.apache.arrow.memory.util.MemoryUtil;
+
+/**
+ * The default Allocation Manager Factory for a module.
+ *
+ * This will be split out to the arrow-memory-netty/arrow-memory-unsafe modules
+ * as part of ARROW-8230. This is currently a placeholder which defaults to Netty.
+ *
+ */
+public class DefaultAllocationManagerFactory implements AllocationManager.Factory {
+
+  public static final AllocationManager.Factory FACTORY = new DefaultAllocationManagerFactory();
+
+  @Override
+  public AllocationManager create(BaseAllocator accountingAllocator, long size) {
+    return new AllocationManager(accountingAllocator) {
+      private final long allocatedSize = size;
+      private final long address = MemoryUtil.UNSAFE.allocateMemory(size);
+
+      @Override
+      public long getSize() {
+        return allocatedSize;
+      }
+
+      @Override
+      protected long memoryAddress() {
+        return address;
+      }
+
+      @Override
+      protected void release0() {
+        MemoryUtil.UNSAFE.freeMemory(address);
+      }
+    };
+  }
+
+  @Override
+  public ArrowBuf empty() {
```

Comment:
Since this is a frequently called method, I think it would be beneficial to reuse the empty ArrowBuf.

Reply:
This is only in tests, but I agree. Fixed